### PR TITLE
Put endpoint

### PR DESCRIPTION
**Pull request by sophialhales**
## Describe Your Changes
- Added functionality to the put endpoint. 
- Added exception handling to the put endpoint.
- Completed testing of put endpoint and exception handling. 
- Created addRule and removeRule methods in Rule entity.
- Created addCondition and removeCondition methods in Condition entity.

## Related Issues
#73

## Type of Change
Please delete options that are not relevant.

- [x] New feature (change that adds functionality)

## Checklist (DOD)
- [x] The code is written and passes all Acceptance criteria's
- [x] Any Zombie code is removed
- [x] Code has proper comment documentation
- [x] The code is unit tested with happy case and passes
- [x] Code is refactored for optimization
- [ ] The code has been peer-reviewed by one or more team members who did not work on it.
- [ ] The code has no merge conflicts

### If Applicable Checklist
- [ ] The code is integration tested and passes (if applicable)
- [ ] The code is regression tested and passes (if applicable)
- [ ] Code is reviewed and accepted by Product Owner (if applicable)

### CODE REVIEW

**Review by Scott-Code-19 — CHANGES_REQUESTED** [review]
Respond/change Comment 1.

File: src/main/java/com/accesspoint/rulesengine/service/RulesetService.java [hunk 1]
```diff
@@ -69,10 +64,84 @@ public ResponseEntity<Ruleset> createRuleset(Ruleset incomingRulesetData) {
...
+            rule.getConditions().stream().forEach(condition -> {
+                if (condition.getFact_type() == null) throw new BadRequestException("Condition fact type cannot be null");
+                if (condition.getValue_type() == null) throw new BadRequestException("Condition value type cannot be null");
+            });
+        });
+
+
+        Ruleset updatedRuleset2 = rulesetRepository.findById(id) //
+                .map(oldRuleset -> {
+                    //TODO: optimize Hibernate select calls
```
Comment (Scott-Code-19) [inline src/main/java/com/accesspoint/rulesengine/service/RulesetService.java:86]: Comment 1:
Remove "TODO" if implemented or explain comment

Comment (GivenTeacher5) [reply src/main/java/com/accesspoint/rulesengine/service/RulesetService.java:86]: needs to be left in for future implementation

**Review by Scott-Code-19 — APPROVED** [review]
Everything looks good and works on my end.
All questions answered

**Pull request merged**